<launch>
<!-- Vehicle Type (0 for jessi, 1 for swervi)-->
    <arg name="model" default="0"/> 
    <arg name="read_from_file" default="true"/>

    <!-- Localization -->
    <include file="$(find igvc_navigation)/launch/localization.launch" />

    <!-- Launch navigation (simulation version) -->
    <include file="$(find igvc_navigation)/launch/mbf_navigation.launch">
        <arg name="model" value="$(arg model)"/>
    </include>

    <!-- Launch back circle service server -->
    <include file="$(find igvc_navigation)/launch/back_circle_server.launch"/>

    <!-- Launch navigation server -->
    <include file="$(find igvc_navigation)/launch/navigation_server.launch"/>

    <!-- Launch navigation client -->
    <include file="$(find igvc_navigation)/launch/navigation_client.launch">
        <arg name="read_from_file" value="$(arg read_from_file)"/>
    </include>

    <!-- Launch drive controller -->
    <group if="$(eval arg('model') == 0)">
        <include file="$(find igvc_navigation)/launch/differential_drive.launch" />
    </group>
    <group if="$(eval arg('model') == 1)">
        <include file="$(find igvc_navigation)/launch/swerve_drive.launch" />
    </group>

    <!-- pointcloud filter -->
    <include file="$(find igvc_perception)/launch/pointcloud_filter.launch" />

    <!-- elevation mapping -->
    <include file="$(find igvc_perception)/launch/elevation_mapping.launch" />

    <!-- traversability filter -->
    <include file="$(find igvc_perception)/launch/slope_filter.launch"/>
</launch>
